pkg/controllers/provisioning/scheduling/node.go

```diff
@@ -27,16 +28,39 @@ import (
 // Node is a set of constraints, compatible pods, and possible instance types that could fulfill these constraints. This
 // will be turned into one or more actual node instances within the cluster after bin packing.
 type Node struct {
-	Constraints         *v1alpha5.Constraints
-	InstanceTypeOptions []cloudprovider.InstanceType
-	Pods                []*v1.Pod
-	unschedulable       bool // if true, the node will accept no more pods
+	Taints               v1alpha5.Taints
+	Requirements         v1alpha5.Requirements
+	Labels               map[string]string
+	InstanceTypeOptions  []cloudprovider.InstanceType
+	Pods                 []*v1.Pod
+	Provider             *v1alpha5.Provider
+	KubeletConfiguration *v1alpha5.KubeletConfiguration
+
+	podResources    v1.ResourceList
+	daemonResources v1.ResourceList
+	daemons         []*v1.Pod
 }
 
-func NewNode(constraints *v1alpha5.Constraints, instanceTypeOptions []cloudprovider.InstanceType, pods ...*v1.Pod) *Node {
-	n := &Node{Constraints: constraints.DeepCopy()}
+func NewNode(constraints *v1alpha5.Constraints, daemons []*v1.Pod, instanceTypeOptions []cloudprovider.InstanceType, pods ...*v1.Pod) *Node {
+	n := &Node{
+		Labels:               constraints.Labels,
+		Taints:               constraints.Taints.DeepCopy(),
+		Requirements:         *constraints.Requirements.DeepCopy(),
```

Comment:
These shouldn't change IIUC, why deep copy?

Reply:
Taints shouldn't, but Requirements will in the `Add()` method